pkg/util/cache/cache.go

```diff
@@ -35,3 +35,40 @@ func BuildAgentKey(keys ...string) string {
 	keys = append([]string{AgentCachePrefix}, keys...)
 	return path.Join(keys...)
 }
+
+// Get returns the value for 'key'.
+//
+// cache hit:
+//
+//	pull the value from the cache and returns it.
+//
+// cache miss:
+//
+//	call 'cb' function to get a new value. If the callback doesn't return an error the returned value is
+//	cached with no expiration date and returned.
+func Get[T any](key string, cb func() (T, error)) (T, error) {
+	return GetWithExpiration[T](key, cb, cache.NoExpiration)
+}
+
+// GetWithExpiration returns the value for 'key'.
+//
+// cache hit:
+//
+//	pull the value from the cache and returns it.
+//
+// cache miss:
+//
+//	call 'cb' function to get a new value. If the callback doesn't return an error the returned value is
+//	cached with the given expire duration and returned.
+func GetWithExpiration[T any](key string, cb func() (T, error), expire time.Duration) (T, error) {
+	if x, found := Cache.Get(key); found {
+		return x.(T), nil
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** Should we check whether the conversion succeeds to avoid unexpected panics ?‏

Reply:
Since we're initializing the cache with the callback that return a `T` value we should be safe. We would panic if the caller init and fetch the same cache key with different type, which should not be done and we have no correct way to react to at runtime.